core/src/main/java/de/javakaffee/web/msm/StorageClientFactory.java

```diff
@@ -78,9 +87,30 @@ static MemcachedClient createCouchbaseClient(final MemcachedNodesManager memcach
             }
         }
 
+        protected static MemcachedNodeURLBasedConnectionFactory createMemcachedNodeURLBasedConnectionFactory() {
+            try {
+                Class.forName("net.spy.memcached.ClientMode");
+                return  Class.forName("de.javakaffee.web.msm.MemcachedElasticConnectionFactory").asSubclass(MemcachedNodeURLBasedConnectionFactory.class).newInstance();
+            } catch (final ClassNotFoundException e) {
+                try {
+                    return Class.forName("de.javakaffee.web.msm.MemcachedConnectionFactory").asSubclass(MemcachedNodeURLBasedConnectionFactory.class).newInstance();
+                } catch (final Exception ex)
+                {
+                    throw new RuntimeException(ex);
+                }
+            } catch (final Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+
         static ConnectionFactory createConnectionFactory(final MemcachedNodesManager memcachedNodesManager,
                                                             final ConnectionType connectionType, final String memcachedProtocol, final String username, final String password, final long operationTimeout,
                                                             final long maxReconnectDelay, final Statistics statistics ) {
+            boolean isClientDynamicMode = false;
+            if (memcachedNodesManager.getMemcachedNodes().contains(".cfg."))
+            {
+                isClientDynamicMode=true;
```

Comment:
Normally the client mode is determined by the node name in aws client. If the node name contains ".cfg." it would be a dynamic mode. Either we pass the Boolean or we pass the node name?
Check the implementation in
aws-elasticache-cluster-client-memcachedfor-java -->
/src/main/java/net/spy/memcached/MemcachedClient.java line
...
if(determineClientMode){
if(addrs.size() == 1){
if(addrs.get(0) == null){
throw new NullPointerException("Socket address is null");
}
String hostName = addrs.get(0).getHostName();
//All config endpoints has ".cfg." subdomain in the DNS name.
if(hostName != null && hostName.contains(".cfg.")){
cf = new DefaultConnectionFactory(ClientMode.Dynamic);
}
}
//Fallback to static mode
if(cf == null){
cf = new DefaultConnectionFactory(ClientMode.Static);
}
...